### PR TITLE
Resolve "1366 expand operation needs to be optimized for large valuesets 3"

### PR DESCRIPTION
**Pull request by dmuylwyk**
See: #1366

### CODE REVIEW

**Comment by dmuylwyk** [comment]
@jamesagnew, please review.

Next up: actually use the terminology tables when expanding.

File: hapi-fhir-jpaserver-base/src/main/java/ca/uhn/fhir/jpa/term/BaseHapiTerminologySvcImpl.java [hunk 1]
```diff
@@ -1491,9 +1495,46 @@ public void storeTermConceptMapAndChildren(ResourceTable theResourceTable, Conce
 		ourLog.info("Done storing TermConceptMap.");
 	}
 
+	@Scheduled(fixedDelay = 600000) // 10 minutes.
+	@Transactional
```
Comment (jamesagnew) [inline hapi-fhir-jpaserver-base/src/main/java/ca/uhn/fhir/jpa/term/BaseHapiTerminologySvcImpl.java]: i'd drop the @tranactional annotation here.. it doesn't do anything but people might mistakenly think it does

Comment (dmuylwyk) [reply hapi-fhir-jpaserver-base/src/main/java/ca/uhn/fhir/jpa/term/BaseHapiTerminologySvcImpl.java]: Done.

File: hapi-fhir-jpaserver-base/src/main/java/ca/uhn/fhir/jpa/entity/TermValueSetExpansionStatusEnum.java [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+
+public enum TermValueSetExpansionStatusEnum {
+
+	NOT_EXPANDED,
```
[the remark this answers is not in the document]
Comment (dmuylwyk) [reply hapi-fhir-jpaserver-base/src/main/java/ca/uhn/fhir/jpa/entity/TermValueSetExpansionStatusEnum.java]: Done.

File: hapi-fhir-jpaserver-base/src/main/java/ca/uhn/fhir/jpa/dao/data/ITermValueSetConceptDao.java [hunk 3]
```diff
@@ -26,10 +26,15 @@
...
+
 public interface ITermValueSetConceptDao extends JpaRepository<TermValueSetConcept, Long> {
 
 	@Query("DELETE FROM TermValueSetConcept vsc WHERE vsc.myValueSet.myId = :pid")
 	@Modifying
-	void deleteTermValueSetConceptsByValueSetId(@Param("pid") Long theValueSetId);
+	void deleteByTermValueSetId(@Param("pid") Long theValueSetId);
+
+	@Query("SELECT vsc FROM TermValueSetConcept vsc WHERE vsc.myValueSet.myId = :pid AND vsc.mySystem = :system_url AND vsc.myCode = :codeval")
+	Optional<TermValueSetConcept> findByValueSetIdSystemAndCode(@Param("pid") Long theValueSetId, @Param("system_url") String theSystem, @Param("codeval") String theCode);
```
[the remark this answers is not in the document]
Comment (dmuylwyk) [reply hapi-fhir-jpaserver-base/src/main/java/ca/uhn/fhir/jpa/dao/data/ITermValueSetConceptDao.java]: Swapped out index for unique constraint, updated migration tasks, and verified that we have logic to ensure a given system|code is added to a given ValueSet only once.

**Pull request merged**